database.hh

```diff
@@ -882,9 +882,15 @@ public:
     void trigger_compaction();
     void try_trigger_compaction() noexcept;
     future<> run_compaction(sstables::compaction_descriptor descriptor);
+
+    // Perform offstrategy compaction in the background
     void trigger_offstrategy_compaction();
+    // Perform offstrategy compaction
     future<> perform_offstrategy_compaction();
+    // Run offstrategy compaction.
+    // Typically called only internally by the compaction manager or unit tests.
```

Comment:
This explanation doesn't explain to me how or why run_offstrategy_compaction() and perform_offstrategy_compaction() differ - they have the same signature.

Reply:
Maybe we should consider moving `run_offstrategy_compaction` to be private and befriend its callers.

That said, I talked with Raphael already about defining a formal abstraction of the table/compaction api so that compaction-manager and compaction would be provided with an object through which they would call back into the table (or test) code, rather than calling them directly on the table.  This way compaction will operate on an abstract container of sstables that could be a table object, but it could also be a TWCS bucket, for example, or just a bunch of sstables in a temporary directory for testing.